crates/bitwarden-crypto/src/keys/symmetric_crypto_key.rs

```diff
@@ -43,21 +56,17 @@ impl SymmetricCryptoKey {
         rng.fill(key.as_mut_slice());
         rng.fill(mac_key.as_mut_slice());
 
-        SymmetricCryptoKey {
-            key,
-            mac_key: Some(mac_key),
-        }
-    }
-
-    pub(crate) fn new(
-        key: Pin<Box<GenericArray<u8, U32>>>,
-        mac_key: Option<Pin<Box<GenericArray<u8, U32>>>>,
-    ) -> Self {
-        Self { key, mac_key }
+        SymmetricCryptoKey::Aes256CbcHmacKey(Aes256CbcHmacKey {
+            encryption_key: key,
+            mac_key,
+        })
     }
 
     fn total_len(&self) -> usize {
-        self.key.len() + self.mac_key.as_ref().map_or(0, |mac| mac.len())
+        match &self {
```

Comment:
I'm quite surprised clippy didn't catch this, but you don't need a `&` on the match value, as self is already received by reference (so you're matching against `&&SymmetricCryptoKey`, which then gets auto dereferenced):
```suggestion
        match self {
```

Same thing below in `to_vec`